src/grpc.ts

```diff
@@ -68,241 +68,242 @@ const COMMON_PROTO_FILES =
         });
 
-/**
- * A class which keeps the context of gRPC and auth for the gRPC.
- *
- * @param {Object=} options - The optional parameters. It will be directly
- *   passed to google-auto-auth library, so parameters like keyFile or
- *   credentials will be valid.
- * @param {Object=} options.auth - An instance of google-auto-auth.
- *   When specified, this auth instance will be used instead of creating
- *   a new one.
- * @param {Object=} options.grpc - When specified, this will be used
- *   for the 'grpc' module in this context. By default, it will load the grpc
- *   module in the standard way.
- * @param {Function=} options.promise - A constructor for a promise that
- * implements the ES6 specification of promise. If not provided, native promises
- * will be used.
- * @constructor
- */
-function GrpcClient(options) {
-  // if (!(this instanceof GrpcClient)) {
-  //   return new GrpcClient(options);
-  // }
-  options = options || {};
-  this.auth = options.auth || autoAuth(options);
-  this.promise = options.promise || Promise;
-  if ('grpc' in options) {
-    this.grpc = options.grpc;
-    this.grpcVersion = '';
-  } else {
-    this.grpc = require('grpc');
-    this.grpcVersion = require('grpc/package.json').version;
-  }
-}
-module.exports = (options) => new GrpcClient(options);
+export class GrpcClient {
+  // tslint:disable-next-line no-any
+  auth: any;
+  promise: PromiseConstructor;
+  // tslint:disable-next-line no-any
+  grpc: any;
+  grpcVersion: string;
 
-/**
- * Creates a gRPC credentials. It asks the auth data if necessary.
- * @private
- * @param {Object} opts - options values for configuring credentials.
- * @param {Object=} opts.sslCreds - when specified, this is used instead
- *   of default channel credentials.
- * @return {Promise} The promise which will be resolved to the gRPC credential.
- */
-GrpcClient.prototype._getCredentials = function _getCredentials(opts) {
-  // tslint:disable-next-line variable-name
-  const PromiseCtor = this.promise;
-  if (opts.sslCreds) {
-    return PromiseCtor.resolve(opts.sslCreds);
+  /**
+   * A class which keeps the context of gRPC and auth for the gRPC.
+   *
+   * @param {Object=} options - The optional parameters. It will be directly
+   *   passed to google-auto-auth library, so parameters like keyFile or
+   *   credentials will be valid.
+   * @param {Object=} options.auth - An instance of google-auto-auth.
+   *   When specified, this auth instance will be used instead of creating
+   *   a new one.
+   * @param {Object=} options.grpc - When specified, this will be used
+   *   for the 'grpc' module in this context. By default, it will load the grpc
+   *   module in the standard way.
+   * @param {Function=} options.promise - A constructor for a promise that
+   * implements the ES6 specification of promise. If not provided, native
+   * promises will be used.
+   * @constructor
+   */
+  constructor(options) {
```

Comment:
Is there a type of `options` here?

Reply:
Yes, but I plan on revisiting this entire file.  I'm trying to do this one chunk at a time :)